### PR TITLE
Pull past purchases

### PR DESCRIPTION
**Pull request by omryYotpo**
@vladshub Could you please review

### CODE REVIEW

File: Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        return;
+      }
+      $token = $this->_api->oauthAuthentication();   
+      if ($token == null) 
+      {                
+        $this->_messageManager->addError(__("Please make sure the APP KEY and SECRET you've entered are correct"));
+        return;
+      }
+      $today = time();
+      $last = $today - (60*60*24*90); //90 days ago
```
Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fix @omryYotpo can you please move this to a config?

File: Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        $this->_messageManager = $messageManager;          
+        $this->_resultRedirectFactory = $resultRedirectFactory;   
+        parent::__construct($context);
+    }
+
+
+    /**
+     * say admin text
+     */
+    public function execute()
```
Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: @omryYotpo who calls this function?

File: Yotpo/Yotpo/Helper/ApiClient.php [hunk 3]
```diff
@@ -102,7 +122,8 @@ public function createApiPost($path, $data, $timeout=self::DEFAULT_TIMEOUT) {
 
   public function createPurchases($order)
   {
-    $this->createApiPost("apps/".$this->_app_key."/purchases", $order);
+    $resData = $this->createApiPost("apps/".$this->_app_key."/purchases", $order);
+    return $resData;
   }
   public function massCreatePurchases($orders, $token)
```
Comment (vladshub) [inline Yotpo/Yotpo/Helper/ApiClient.php]: @omryYotpo why not just return to the function if you don't use the variable?

File: Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php (hunk 2 above)
Comment (omryYotpo) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: The controller's execute function is called when the button is clicked and then an ajax request calls the controller's execute

**Comment by omryYotpo** [comment]
@vladshub  could you please re-review?

File: Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php (hunk 1 above)
Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fixed

Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fix

Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fixed

**Comment by vladshub** [comment]
@omryYotpo Please rebase this
@NetanelMoshkovitz this is fine.

File: Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php (hunk 1 above)
Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fix

Comment (vladshub) [inline Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php]: !fixed

**Comment by omryYotpo** [comment]
@dmitri86git fixed https://jira.yotpo.com/browse/YO-1160 
pls check

**Comment by dmitri86git** [comment]
@omryYotpo @j-bachman Please rebase master an it can be merged :bomb:

**Comment by omryYotpo** [comment]
@vladshub Rebased...I need a star

File: Yotpo/Yotpo/Block/Config.php [hunk 4]
```diff
@@ -25,6 +25,11 @@ public function getSecret()
         return $this->_scopeConfig->getValue(self::YOTPO_SECRET, ScopeInterface::SCOPE_STORE);
     }
 
+    public function getCostumeOrderStatus()
+    {
+        null;
```
Comment (vladshub) [inline Yotpo/Yotpo/Block/Config.php]: @omryYotpo why do we need this function?

File: Yotpo/Yotpo/etc/adminhtml/system.xml [hunk 5]
```diff
@@ -20,43 +20,53 @@
                     <p>Click <a href="http://support.yotpo.com/entries/22635391-advanced-magento-features" target="_blank" 
                       title="yotpo">here</a> for the advanced customization guide. 
                     </p><br/>
-                    <p style="font-size:18px;">Dont have an App Key? <a style="font-size:18px;" href="http://my.yotpo.com/" target="_blank" 
+                    <p style="font-size:18px;">Dont have an App Key? <a style="font-size:18px;" href="https://my.yotpo.com/register?utm_source=customers_magento_admin&utm_medium=link&utm_campaign=magento_admin_signin_link" target="_blank"
```
Comment (vladshub) [inline Yotpo/Yotpo/etc/adminhtml/system.xml]: @billykov & @omryYotpo 
We can have here a one click registration button ...

File: Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?php
+?>
+
+<script type="text/javascript">
+    //<![CDATA[
+    function exportOrders() {debugger;
+    var params = {
+                form_key : FORM_KEY,
```
Comment (vladshub) [inline Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml]: @omryYotpo what does "FORM_KEY" means?

File: Yotpo/Yotpo/Block/Config.php (hunk 4 above)
Comment (omryYotpo) [inline Yotpo/Yotpo/Block/Config.php]: @vladshub For when we'll want to implement an option for custom order status

File: Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml (hunk 6 above)
Comment (omryYotpo) [inline Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml]: It's a magento variable that is random and used for each session, you need to pass it when making an Ajax request

File: Yotpo/Yotpo/etc/adminhtml/system.xml (hunk 5 above)
Comment (billykov) [inline Yotpo/Yotpo/etc/adminhtml/system.xml]: @vladshub Yes, there is a different task for it.

File: Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml (hunk 6 above)
Comment (vladshub) [inline Yotpo/Yotpo/view/adminhtml/templates/system/config/yotpobutton.phtml]: @omryYotpo :cool:

**Comment by vladshub** [comment]
:star:

**Pull request merged**